cdhweb/static_src/global/rich-text.scss

```diff
@@ -1,56 +1,95 @@
-// Styles for body copy, aka "rich text".
-// Wrap body copy elements in a div/section with the class "rich-text".
-// Works best if the direct children are the typographic elements
-// (headings, p etc). This ensures that the .rich-text container doesn't
-// have extra space at the start/end.
-// :where() selector is used to keep specificity low.
-:where(.rich-text) {
-  --space-between-els: 1lh; // relative so it scales to bigger/smaller text
-  --space-between-sections: #{px2rem(40)};
-
-  @include xl {
-    --space-between-sections: #{px2rem(64)};
-  }
-
-  > * {
-    margin-bottom: var(--space-between-els, 1em);
-  }
-
-  // When we encounter a heading, that can be considered a new "section" of
-  // the content so should have a nice big space above...
-  // Note: h1s are assumed to be outside of the `rich-text` area, as they're
-  // typically in a hero banner rather than coming from a WYSIWYG with other text.
-  :is(h2, h3, h4, h5, h6) {
-    margin-top: var(--space-between-sections);
-  }
-  // ...unless a heading appears immediately after another heading, in which case we need to reduce the spacing.
-  h2 + h3,
-  h3 + h4,
-  h4 + h5,
-  h5 + h6 {
-    margin-top: var(--space-between-els, 1em);
-  }
-
-  // Remove any orphan vertical space at the start/end of the rich-text block.
-  // NOTE: this assumes items are direct children of .rich-text container.
-  > * {
-    &:first-child {
-      margin-top: 0;
-    }
-    &:last-child {
-      margin-bottom: 0;
-    }
-  }
-
-  ul,
-  ol {
-    // Undo padding:0 from the css reset, which kills list indentation.
-    padding: revert;
-  }
-
-  // Collapsing margins should ensure we don't get extra space between this and preceding element.
-  // i.e. no `:not(:first...)` required.
-  li {
-    margin-top: 0.5em;
+// By default, every prose element has a bottom margin.
+// `li` is excluded because list item spacing is handled later on.
+// Same with headings because they have their own, more specific spacing values.
+// Ignore header and footer, they have their own styles.
+:where(main) {
+  --space-default: 1rem;
+
+  --space-l2-lg: #{px2rem(56)};
+  --space-l3-lg: #{px2rem(40)};
+  --space-l4-lg: #{px2rem(24)};
+  --space-l5-lg: #{px2rem(40)};
+  --space-l6-lg: #{px2rem(40)};
```

Comment:
I would have expected `space-l5` and `space-l6` to be the smaller than `space-l4` — how come they're not?

Reply:
There are no styles for h5 or h6. I've given them the same font-size etc too for now (which is not ideal, but a better fallback than them being bigger than h4). H4 is already the same size as `p` text, and i felt funny about making headings smaller than `p` text :/
<img width="723" alt="Screenshot 2024-05-20 at 4 05 08 PM" src="https://github.com/springload/cdh-web/assets/1134713/59a7ce6e-db0f-4c12-b71d-c7c52fb54521">

Reply:
you could make the spacing for both `#{px2rem(24)}` so that they're same as the `l4` for now as well